tests/db_functions/datetime/test_extract_trunc.py

```diff
@@ -940,14 +940,18 @@ def test_trunc_lookup_name_sql_injection(self):
             end_datetime = timezone.make_aware(end_datetime)
         self.create_model(start_datetime, end_datetime)
         self.create_model(end_datetime, start_datetime)
-        msg = "Invalid kind: "
-        with self.assertRaisesMessage(ValueError, msg):
-            DTModel.objects.filter(
+        # Database backends raise an exception or don't return any results.
+        try:
+            exists = DTModel.objects.filter(
                 start_datetime__date=Trunc(
                     "start_datetime",
                     "year', start_datetime)) OR 1=1;--",
                 )
             ).exists()
+        except (DataError, OperationalError):
```

Comment:
newb here, I'm trying to understand why you are catching both `DataError` and `OperationalError`

I'm guessing this test is somehow run on all 3 backends and they raise different exceptions?

_same with your [other change in this file](https://github.com/django/django/pull/15820/files#diff-7704d1b7fbfb3516a17d0f19338242c16eff438b3e06125015b35d62a314f638R248):_

```py
with self.assertRaises((DataError, OperationalError, ValueError)):
```

Reply:
Yes.